### PR TITLE
Manytomany

### PR DESCRIPTION
**Pull request by Blackvals**
Changes including Allergy entity plus some refactors.

### CODE REVIEW

File: src/SeedCppCore/DAL/Translators/JSON/PatientJSONTranslator.cpp [hunk 1]
```diff
@@ -21,7 +21,12 @@ namespace seed_cpp { namespace dal {
 		jsonPatient.addMember("surname", m_patient.getSurname());
 		jsonPatient.addMember("name", m_patient.getName());
 		jsonPatient.addMember("email", m_patient.getEmail());
-		jsonPatient.addMember("dob", boost::posix_time::to_iso_string(m_patient.getDob()));
+		
+		auto dob = m_patient.getDob();
+		if (dob)
+		{
+			jsonPatient.addMember("dob", boost::posix_time::to_iso_string(*m_patient.getDob()));
```
Comment (joaquimvila) [inline src/SeedCppCore/DAL/Translators/JSON/PatientJSONTranslator.cpp]: I would use the _boost::posix_time::from_iso_extended_string_ method to fix the compatibility issue for the dates of the API. This issue is already fixed in unittests branch.

File: src/SeedCppCore/Services/System/ResourceService.cpp [hunk 2]
```diff
@@ -9,23 +9,41 @@
...
-	const std::string ResourceService::schemaModelPatient = "Model/Patient.json";
-	const std::string ResourceService::schemaModelAddress = "Model/Address.json";
-	const std::string ResourceService::schemaModelUser = "Model/User.json";
+	// Endpoint schemas
+	const std::string ResourceService::schemaPatientPost	= "Endpoint/PatientsPostEndpoint.json";
+	const std::string ResourceService::schemaPatientPut		= "Endpoint/PatientsPutEndpoint.json";
+	const std::string ResourceService::schemaUserPost		= "Endpoint/UsersPostEndpoint.json";
+	const std::string ResourceService::schemaUserPut		= "Endpoint/UsersPutEndpoint.json";
+	const std::string ResourceService::schemaAllergyPost	= "Endpoint/AllergiesPostEndpoint.json";
+	const std::string ResourceService::schemaAllergyPut		= "Endpoint/UsersPutEndpoint.json";
```
Comment (joaquimvila) [inline src/SeedCppCore/Services/System/ResourceService.cpp]: Path to the allergies PUT endpoint is not correct

File: src/SeedCppCore/DAL/Translators/JSON/PatientJSONTranslator.cpp (hunk 1 above)
Comment (joaquimvila) [inline src/SeedCppCore/DAL/Translators/JSON/PatientJSONTranslator.cpp]: I would use the boost::posix_time::from_iso_extended_string method to fix the compatibility issue for the dates of the API. This issue is already fixed in unittests branch.

**Review by joaquimvila — CHANGES_REQUESTED** [review]

File: src/SeedCppCore/Services/System/ResourceService.cpp (hunk 2 above)
Comment (Blackvals) [reply src/SeedCppCore/Services/System/ResourceService.cpp]: Copy-paste error I missed. Fixed now.

File: src/SeedCppCore/DAL/Translators/JSON/PatientJSONTranslator.cpp (hunk 1 above)
Comment (Blackvals) [reply src/SeedCppCore/DAL/Translators/JSON/PatientJSONTranslator.cpp]: Done for both save and load translator

**Review by joaquimvila — APPROVED** [review]

**Pull request merged**